### PR TITLE
Use ScenarioPropertiesConfigFinalizer in NewProcessPreparer and ScenarioPropertiesValidatior

### PR DESCRIPTION
**Pull request by wrzontek**
## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/definition/ScenarioPropertiesConfigFinalizer.scala [hunk 1]
```diff
@@ -12,7 +12,12 @@ class ScenarioPropertiesConfigFinalizer(
   def finalizeScenarioProperties(
       scenarioPropertiesConfig: Map[String, ScenarioPropertyConfig],
   ): Map[String, ScenarioPropertyConfig] = {
-    additionalUIConfigProvider.getScenarioPropertiesUIConfigs(processingType) |+| scenarioPropertiesConfig
+    additionalUIConfigProvider
+      .getScenarioPropertiesUIConfigs(processingType)
+      .filter { case (propertyName, _) =>
+        // configs from additionalUIConfigProvider should only override existing configs, not create new ones
```
Comment (arkadius) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/definition/ScenarioPropertiesConfigFinalizer.scala:18]: Why?

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/server/AkkaHttpBasedRouteProvider.scala [hunk 2]
```diff
@@ -200,7 +200,8 @@ class AkkaHttpBasedRouteProvider(
       val newProcessPreparer = processingTypeDataProvider.mapValues { processingTypeData =>
         new NewProcessPreparer(
           processingTypeData.deploymentData.metaDataInitializer,
-          processingTypeData.deploymentData.scenarioPropertiesConfig
+          processingTypeData.deploymentData.scenarioPropertiesConfig,
```
Comment (arkadius) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/server/AkkaHttpBasedRouteProvider.scala:203]: Why can't we just somehow merge `deploymentData.scenarioPropertiesConfig` with these from `additionalUIConfigProvider` at the very beginning of the process in the place where they are already merged with properties from config - the same as we do with component's parameters?

Comment (wrzontek) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/server/AkkaHttpBasedRouteProvider.scala:203]: I guess we could, but this way changes to the return value of `getScenarioPropertiesUIConfigs` take effect immediately, without requiring model reload.

Comment (wrzontek) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/server/AkkaHttpBasedRouteProvider.scala:203]: wait, let me check some things

Comment (wrzontek) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/server/AkkaHttpBasedRouteProvider.scala:204]: i've managed to rather easily get rid of this limitation

>   // `ScenarioPropertyConfig.validators` currently does nothing (only usage goes to createUIScenarioPropertyConfig)

including them in ScenarioPropertiesValidator pretty much makes them equal to `normal` scenario properties, so i've also removed the filter from the thread above

**Review by arkadius — APPROVED** [review]

**Review by ForrestFairy — APPROVED** [review]
Let's merge it

**Comment by wrzontek** [comment]
nondeterministic cypress tests failure
check out the cypress PR's, they change the same file back and forth

not fault of this PR so i'll merge it

**Pull request merged**